Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseTimeExtractor.java

```diff
@@ -19,14 +28,120 @@ public String getExtractorName() {
     public BaseTimeExtractor(ITimeExtractorConfiguration config) {
         this.config = config;
     }
-
     @Override
     public List<ExtractResult> extract(String input, LocalDateTime reference) {
-        throw new UnsupportedOperationException();
+        List<Token> tokens = new ArrayList<>();
+        tokens.addAll(BasicRegexMatch(input));
+        tokens.addAll(AtRegexMatch(input));
+        tokens.addAll(BeforeAfterRegexMatch(input));
+        tokens.addAll(SpecialCasesRegexMatch(input));
+
+        List<ExtractResult> timeErs = Token.mergeAllTokens(tokens, input, getExtractorName());
+        if (this.config.getOptions() == DateTimeOptions.EnablePreview)
+        {
+            timeErs = MergeTimeZones(timeErs, config.getTimeZoneExtractor().extract(input, reference), input);
+        }
+        return timeErs;
     }
 
     @Override
     public List<ExtractResult> extract(String input) {
         return this.extract(input, LocalDateTime.now());
     }
+
+    private List<ExtractResult> MergeTimeZones(List<ExtractResult> timeErs, List<ExtractResult> timeZoneErs, String text)
+    {
```

Comment:
NIT: Code standard. Curly brace should be on the same line as the declaration. 👮